24730 sa2 revision

SebVortex opened the pull request:
## Description
Need to make direct deposit input fields required and add an "I don't want to use direct deposit" checkbox to the bank account page of 1990e.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#24730

## Testing done
local testing

## Screenshots
<img width="518" alt="Screen Shot 2022-04-07 at 4 04 50 PM" src="https://user-images.githubusercontent.com/86262790/162309814-076e6e0e-911b-4a9e-949d-fa8a4c821b86.png">

<img width="469" alt="Screen Shot 2022-04-07 at 4 09 27 PM" src="https://user-images.githubusercontent.com/86262790/162310600-ec65c46e-289d-47fd-b0e1-1534aad1c512.png">



## Acceptance criteria
- [x] account type is required
- [x] account number is required
- [x] routing number is required
- [x] I don't want to use direct deposit checkbox is present
- [x] checkbox hides bankAccount input fields

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

SebVortex committed:
Added custom presubmitInfo for review portion of form

Also removed unused FormFooter

SebVortex committed:
Re-applied FormFooter

SebVortex committed:
Remade preSubmitInfo to be more inline with old presubmitInfo

including more data than just a presubmit message in the schema

SebVortex committed:
Description now showing on direct deposit portion

SebVortex committed:
Additional info now showing on direct deposit form

Also using va-additional-info instead of Additional Info

SebVortex committed:
Started sa2 revision

SebVortex committed:
Merge branch '24730-part-2' into 24730-sa2-revision

SebVortex committed:
requireds now showing

SebVortex committed:
decline Direct Deposit checkbox implemented

SebVortex committed:
merge with using direct deposit

SebVortex committed:
fields are now required only if direct-deposit isn't declined

SebVortex committed:
Merge branch 'master' into 24730-sa2-revision

SebVortex committed:
Fixed issue with formatting and linting from resolving merge conflict

SebVortex committed:
Addressed 'import/no-named-as-default-member'

SebVortex committed:
Fixed react error. Used merge to safely incorporate view elements without throwing an error

SebVortex committed:
Removed unused uiSchema

nbarthelemy reviewed (approved):
Nice job! @SebVortex

SebVortex committed:
Removed empty required array from bank account properties

SebVortex committed:
removed required error messages

SebVortex committed:
changed widgetClassNames to classNames

SebVortex committed:
Cleaned up uiSchema in merge and changed pageTests to properly test bank Account forms with different amounts of inputs

Also fixed bug where null/undefined declineDirectDeposit causes a react error in unit tests

SebVortex committed:
adjusted top margin of 'I don't want to use direct deposit' checkbox

SebVortex committed:
removed unused schema import for directDeposit
